src/types/shortcuts.ts

```diff
@@ -1,22 +1,25 @@
+import { z, ZodObject, ZodRawShape } from 'zod'
 import { NetworkId } from './networkId'
 
 export interface ShortcutsHook {
   getShortcutDefinitions(
     networkId: NetworkId,
     address?: string,
-  ): Promise<ShortcutDefinition[]>
+  ): Promise<ShortcutDefinition<any>[]>
 }
 
-export interface ShortcutDefinition {
+export interface ShortcutDefinition<TriggerInputShape extends ZodRawShape> {
   id: string // Example: claim-reward
   name: string // Example: Claim
   description: string // Example: Claim your reward
   networkIds: NetworkId[] // Example: ['celo-mainnet']
   category?: 'claim' // We'll add more categories later
+  triggerInputShape: TriggerInputShape // Zod object shape of the input for the trigger
   onTrigger: (
-    networkId: NetworkId,
-    address: string,
-    positionAddress: string,
+    args: {
+      networkId: NetworkId
+      address: string
+    } & z.infer<ZodObject<TriggerInputShape>>,
```

Comment:
This is the main interesting change.

The additional trigger arguments are fully checked at runtime + compile time thanks to Zod.